meilisearch-http/src/task.rs

```diff
@@ -39,10 +40,29 @@ impl From<TaskContent> for TaskType {
     }
 }
 
+#[derive(Debug)]
+pub struct TaskTypeError {
+    invalid_type: String,
+}
+
+impl fmt::Display for TaskTypeError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "invalid task type `{}`, expecting one of: \
+            indexCreation, indexUpdate, indexDeletion, documentAdditionOrUpdate, \
+            documentDeletion, settingsUpdate, dumpCreation",
+            self.invalid_type
+        )
+    }
+}
+
+impl Error for TaskTypeError {}
+
 impl FromStr for TaskType {
-    type Err = String;
+    type Err = TaskTypeError;
 
-    fn from_str(status: &str) -> Result<Self, String> {
+    fn from_str(status: &str) -> Result<Self, TaskTypeError> {
```

Comment:
Could you change the name of this parameter to be named `type_`, please? It has been copy/pasted from the `TaskStatus` impl by myself and I forgot to change the name :)

Reply:
done!